src/idl_gen_dlang.cpp

```diff
@@ -0,0 +1,706 @@
+   void GenComment(const std::vector<std::string> &dc, std::string *code_ptr, const char *prefix = "") {
+       if (dc.begin() == dc.end()) {
+           // Don't output empty comment blocks with 0 lines of comment content.
+           return;
+       }
+
+       std::string &code = *code_ptr;
+
+       std::string line_prefix = std::string(prefix) + "///";
+       for (auto it = dc.begin();
+            it != dc.end();
+            ++it) {
+           code += line_prefix + *it + "\n";
+       }
+   }
+
+
+   bool SaveType(const std::string &defname, const Namespace &ns
+                 , const std::string &classcode, bool needs_includes) {
+
+       if (!classcode.length()) return true;
+
+       std::string code;
+       code = code + "// " + FlatBuffersGeneratedWarning();
+       code += "module ";
+       std::string namespace_name = FullNamespace(".", ns);
+        if (!namespace_name.empty()) {
+            code += namespace_name + ".";
+            code += defname + ";\n\n";
+        }
+        if (needs_includes)
+            code += "import std.typecons;\nimport flatbuffers;\n\n";
+       code += classcode;
+       auto filename = NamespaceDir(ns) + defname + ".d";
+       return SaveFile(filename.c_str(), code, false);
+   }
+
+
+   // Returns the function name that is able to read a value of the given type.
+   std::string GenGetter( const Type &type) {
+       switch (type.base_type) {
+       case BASE_TYPE_STRING: return "__string";
+       case BASE_TYPE_STRUCT: return "__struct";
+       case BASE_TYPE_UNION:  return "__union!";
+       case BASE_TYPE_VECTOR: return GenGetter(type.VectorType());
+       default: {
+           std::string getter = std::string("_buffer.") + "get!";
+           if (type.base_type == BASE_TYPE_BOOL) {
+               getter += "ubyte";
+           } else {
+               getter += GenTypeGet(type);
+           }
+           return getter;
+       }
+       }
+   }
+
+   // Returns the method name for use with add/put calls.
+   std::string GenMethod( const Type &type) {
+       return IsScalar(type.base_type)
+               ? (std::string("!") + GenTypeBasic( type))
+               : (IsStruct(type) ? "Struct" : "Offset");
+   }
+
+   // Recursively generate arguments for a constructor, to deal with nested
+   // structs.
+   void GenStructArgs(const StructDef &struct_def,
+                             std::string *code_ptr, const char *nameprefix) {
+       std::string &code = *code_ptr;
+       for (auto it = struct_def.fields.vec.begin();
+            it != struct_def.fields.vec.end();
+            ++it) {
+           auto &field = **it;
+           if (IsStruct(field.value.type)) {
+               // Generate arguments for a struct inside a struct. To ensure names
+               // don't clash, and to make it obvious these arguments are constructing
+               // a nested struct, prefix the name with the struct name.
+               GenStructArgs( *field.value.type.struct_def, code_ptr,
```

Comment:
redundant space after `(`, also might need fixing alignment in the next line to match.